File: src/p0898/java/solution.java
import java.util.HashSet;

class Solution {
    public int subarrayBitwiseORs(int[] nums) {
        HashSet<Integer> result = new HashSet<>();
        HashSet<Integer> prev = new HashSet<>();
        for (int num : nums) {
            prev.add(0);
            HashSet<Integer> curr = new HashSet<>();
            for (int x : prev) {
                curr.add(x | num);
                result.add(x | num);
            }
            prev = curr;
        }
        return result.size();
    }
}

File: src/p0898/cpp/solution.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
public:
    int subarrayBitwiseORs(vector<int> &nums) {
        unordered_set<int> result;
        unordered_set<int> s;
        for (auto num : nums) {
            s.insert(0);
            unordered_set<int> t;
            for (auto x : s) {
                t.insert(x | num);
                result.insert(x | num);
            }
            s.swap(t);
        }
        return result.size();
    }
};
